src/osemgrep/language_server/LS_e2e.ml

```diff
@@ -0,0 +1,891 @@
+(* Brandon Wu
+ *
+ * Copyright (C) 2019-2023 Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
```

Comment:
If this is mostly a port of test_ls.py, then at least say so in a toplevel comment, and maybe how much coverage you got compared to test_ls.py, and when we can get rid of test_ls.py, etc.